brokerapi/brokers/cloudsql/broker.go

```diff
@@ -314,10 +314,11 @@ func (b *CloudSQLBroker) finishProvisioning(instanceId string, params map[string
 	// it should be quick and not actually async.
 	var err error
 
-	instance := models.ServiceInstanceDetails{}
-	if err = db_service.DbConnection.Where("ID = ?", instanceId).First(&instance).Error; err != nil {
+	instancePtr, err := db_service.GetServiceInstanceDetailsById(instanceId)
+	if err != nil {
 		return brokerapi.ErrInstanceDoesNotExist
 	}
+	instance := *instancePtr
```

Comment:
I see a couple of places where you duplicate the struct. I'm assuming this is to ensure the same functionality as before.

However, I think it's not necessary. If you follow this code, you can see that even if called sub-functions edit the struct, the final struct passed to the save call is still the copy. So you should be safe to just pass the pointer around.